### PR TITLE
Ontology_HD-Map

### PR DESCRIPTION
**Pull request by jtdemer**
The ontology was created on 02/27/2014 with the latest version of the Ontology Creator and uses the current Excel file as input: Metadata, which was created in collaboration with all data providers from the GAIA-X4PLCC-AAD project and can now be merged into the main branch.

Signed-off-by: jtdemer <johannes.demer@asc-s.de>

### CODE REVIEW

**Review by 3Dbastian — DISMISSED** [review]
did not check it explicitly, but approved for workflow testing reasons, 
there might be a new version within the map-and-scenario repo the next days

**Review by JustinDxx — CHANGES_REQUESTED** [review]

**Comment by lenasauermann** [comment]
General Ontology should be linked

File: HD-Map/hdmap_shacl.ttl [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+            sh:minCount 1 ;
+            sh:node hdmap:QualityShape ;
+            sh:order 1 ;
+            sh:path hdmap:Quality ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node hdmap:GeoreferenceShape ;
+            sh:order 7 ;
+            sh:path hdmap:Georeference ] ;
+    sh:targetClass hdmap:hdmap .
```
Comment (robertschubert) [inline HD-Map/hdmap_shacl.ttl:42]: Dont forget to change the targetClass to HdMap when ontology is changed.

File: HD-Map/hdmap_shacl.ttl [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+            sh:description "Format type definition"@en ;
+            sh:in ("ASAM OpenDRIVE" "Lanelet" "Road5" "Shape" "road2sim" "roadXML") ;
+            sh:maxCount 1 ;
+            sh:message "Validation of type failed!"@en ;
+            sh:name "type"@en ;
+            sh:order 0 ;
+            sh:path hdmap:type ] ;
+    sh:targetClass hdmap:Format .
+
+hdmap:GeneralShape a sh:NodeShape ;
```
Comment (robertschubert) [inline HD-Map/hdmap_shacl.ttl:109]: There is also a "general:GeneralShape" defined by Mirco (https://github.com/GAIA-X4PLC-AAD/ontology-management-base/pull/27)
This one seems to be something different. Is "General" the right name for it?

As I understood Mirco all DataShapes (like HdMap) should include the `general:GeneralShape`. Did you forget it to include?

File: HD-Map/hdmap_ontology.ttl [hunk 3]
```diff
@@ -0,0 +1,18 @@
+@prefix dcterms: <http://purl.org/dc/terms/> .
+@prefix gax-core: <https://w3id.org/gaia-x/core#> .
+@prefix hdmap: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/hdmap/> .
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix gx: <https://registry.lab.gaia-x.eu/development/api/trusted-shape-registry/v1/shapes/jsonld/trustframework#> .
+
+hdmap: a owl:Ontology ;
```
[the remark this answers is not in the document]
Comment (3Dbastian) [reply HD-Map/hdmap_ontology.ttl]: fixed

File: HD-Map/hdmap_ontology.ttl [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    rdfs:label "ontology definition for hdmap"@en ;
+    dcterms:contributor "Sebastian Tuttas (3DM)" ;
+    owl:versionInfo "0.1"^^xsd:float .
+
+hdmap:hdmap a owl:Class ;
```
[the remark this answers is not in the document]
Comment (3Dbastian) [reply HD-Map/hdmap_ontology.ttl]: fixed

File: HD-Map/hdmap_shacl.ttl [hunk 5]
```diff
@@ -0,0 +1,401 @@
+@prefix hdmap: <https://github.com/GAIA-X4PLC-AAD/ontology-management-base/tree/main/hdmap/> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix skos: <http://www.w3.org/2004/02/skos/core#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+hdmap:hdmapShape a sh:NodeShape ;
```
[the remark this answers is not in the document]
Comment (3Dbastian) [reply HD-Map/hdmap_shacl.ttl]: fixed

File: HD-Map/hdmap_ontology.ttl [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    rdfs:label "class definition for hdmap" ;
+    rdfs:comment "Attributes for high definition maps"@en ;
+    rdfs:subClassOf gax-core:DataResource .
```
[the remark this answers is not in the document]
Comment (3Dbastian) [reply HD-Map/hdmap_ontology.ttl]: fixed

File: HD-Map/hdmap_shacl.ttl [hunk 7]
```diff
@@ -0,0 +1,401 @@
...
+    sh:property [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node hdmap:FormatShape ;
+            sh:order 5 ;
+            sh:path hdmap:Format ],
```
[the remark this answers is not in the document]
Comment (3Dbastian) [reply HD-Map/hdmap_shacl.ttl]: fixed

File: HD-Map/hdmap_shacl.ttl (hunk 1 above)
Comment (3Dbastian) [reply HD-Map/hdmap_shacl.ttl]: fixed

File: hdmap/hdmap_shacl.ttl [hunk 8]
```diff
@@ -349,7 +215,7 @@ hdmap:QuantityShape a sh:NodeShape ;
             sh:maxCount 1 ;
             sh:message "Validation of speedLimit failed!"@en ;
             sh:name "speedLimit"@en ;
-            sh:node hdmap:Range2DShape ;
+            sh:node general:Range2DShape ;
```
Comment (robertschubert) [inline hdmap/hdmap_shacl.ttl:218]: as of today 2 pm there is no Range2DShape in general. 
Will it be added or should it be reintroduced here (see line 389 in the diff view on the left)

File: hdmap/hdmap_shacl.ttl [hunk 9]
```diff
@@ -38,7 +40,7 @@ hdmap:HdMapShape a sh:NodeShape ;
             sh:minCount 1 ;
             sh:node hdmap:GeoreferenceShape ;
             sh:order 7 ;
-            sh:path hdmap:georeference ] ;
+            sh:path georeference:georeference ] ;
```
Comment (robertschubert) [inline hdmap/hdmap_shacl.ttl:43]: ti should be the other way round.
`sh:path` should be `hdmap:georeference`
`sh:node` should be `georeference:GeoreferenceShape`

File: hdmap/hdmap_shacl.ttl [hunk 10]
```diff
@@ -124,143 +126,7 @@ hdmap:GeneralShape a sh:NodeShape ;
...
-        [ skos:example "Autobahn A8, Großraum Ingolstadt" ;
-            sh:datatype xsd:string ;
-            sh:description "Description of the location, dependen if it is a more trajectory style data or area style data"@en ;
-            sh:message "Validation of relationOrArea failed!"@en ;
-            sh:maxCount 1 ;
-            sh:name "relationOrArea"@en ;
-            sh:order 4 ;
-            sh:path hdmap:relationOrArea ] ;
-    sh:targetClass hdmap:ProjectLocation .     
+    sh:targetClass hdmap:General .
```
Comment (robertschubert) [inline hdmap/hdmap_shacl.ttl:129]: is this `GeneralShape` different from the `GeneralShape` in `general`?

**Review by robertschubert — CHANGES_REQUESTED** [review]

File: HD-Map/HD-Map_ontology.ttl [hunk 11]
```diff
@@ -0,0 +1,17 @@
+@prefix HD-Map: <https://github.com/GAIA-X4PLC-AAD/map-and-scenario-data/HD-Map_ontology/> .
```
[the remark this answers is not in the document]
Comment (MircoNierenz) [reply HD-Map/HD-Map_ontology.ttl]: outdated

File: HD-Map/HD-Map_shacl.ttl [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+            sh:path HD-Map:HD-Map_general_recording_time ],
+        [ skos:example "EPSG 32632" ;
+            sh:datatype xsd:string ;
+            sh:description "EPSG code of the map"@en ;
+            sh:maxCount "1"^^xsd:unsignedInt ;
+            sh:message "Validation of projection_type failed!"@en ;
+            sh:minCount "1"^^xsd:unsignedInt ;
+            sh:name "projection_type"@en ;
+            sh:path HD-Map:HD-Map_Georefencing_projection_type ] ;
+    sh:targetClass HD-Map:Asset .
```
[the remark this answers is not in the document]
Comment (MircoNierenz) [reply HD-Map/HD-Map_shacl.ttl]: outdated

File: HD-Map/HD-Map_ontology.ttl [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+@prefix owl: <http://www.w3.org/2002/07/owl#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+HD-Map: a owl:Ontology ;
+    rdfs:label "ontology definition for HD-Map"@en ;
+    dcterms:contributor "Sebastian Tuttas (3DM)" ;
+    owl:versionInfo "0.1"^^xsd:float .
+
+HD-Map:HD-Map a owl:Class ;
```
[the remark this answers is not in the document]
Comment (MircoNierenz) [reply HD-Map/HD-Map_ontology.ttl]: outdated

File: HD-Map/HD-Map_shacl.ttl [hunk 14]
```diff
@@ -0,0 +1,196 @@
+@prefix HD-Map: <https://github.com/GAIA-X4PLC-AAD/map-and-scenario-data/HD-Map_ontology> .
```
[the remark this answers is not in the document]
Comment (MircoNierenz) [reply HD-Map/HD-Map_shacl.ttl]: outdated

File: hdmap/hdmap_shacl.ttl (hunk 10 above)
Comment (MircoNierenz) [reply hdmap/hdmap_shacl.ttl:130]: yes is different
Todo - better naming

File: hdmap/hdmap_shacl.ttl (hunk 9 above)
Comment (MircoNierenz) [reply hdmap/hdmap_shacl.ttl:44]: done

File: hdmap/hdmap_shacl.ttl (hunk 8 above)
Comment (MircoNierenz) [reply hdmap/hdmap_shacl.ttl:219]: add Range2DShape

Comment (MircoNierenz) [reply hdmap/hdmap_shacl.ttl:219]: Question: this is a very general datenstruct and can be used in other shacl's. Better then in General?

File: HD-Map/hdmap_shacl.ttl (hunk 2 above)
Comment (MircoNierenz) [reply HD-Map/hdmap_shacl.ttl]: please discuss in the meeting

File: hdmap/hdmap_shacl.ttl [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+            sh:order 5 ;
+            sh:path hdmap:format ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node hdmap:ContentShape ;
+            sh:order 2 ;
+            sh:path hdmap:content ],
+        [ sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:node hdmap:GeneralShape ;
```
Comment (robertschubert) [inline hdmap/hdmap_shacl.ttl:22]: Test

**Review by robertschubert — CHANGES_REQUESTED** [review]

File: hdmap/hdmap_shacl.ttl (hunk 15 above)
Comment (robertschubert) [reply hdmap/hdmap_shacl.ttl:22]: task resolved

File: HD-Map/hdmap_shacl.ttl (hunk 2 above)
Comment (MircoNierenz) [reply HD-Map/hdmap_shacl.ttl]: local general node was removed and generalShape was integrated. done!

File: hdmap/hdmap_shacl.ttl (hunk 8 above)
Comment (MircoNierenz) [reply hdmap/hdmap_shacl.ttl:219]: Range2DShape moved to general and used in hdmap. done!

File: hdmap/hdmap_shacl.ttl (hunk 10 above)
Comment (MircoNierenz) [reply hdmap/hdmap_shacl.ttl:130]: remove local general node. done!

File: hdmap/hdmap_ontology.ttl [hunk 16]
```diff
@@ -0,0 +1,17 @@
+@prefix dcterms: <http://purl.org/dc/terms/> .
```
Comment (robertschubert) [inline hdmap/hdmap_ontology.ttl:1]: the _instance.json file is missing. 
The rest looks good now.

**Review by robertschubert — CHANGES_REQUESTED** [review]

File: hdmap/hdmap_ontology.ttl (hunk 16 above)
Comment (MircoNierenz) [reply hdmap/hdmap_ontology.ttl:1]: add hdmap-instance.json. done!

**Review by robertschubert — APPROVED** [review]
approved

**Review by 3Dbastian — APPROVED** [review]
initial version for main branch approved

File: hdmap/hdmap-instance.json [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+      "driving",
+      "walking"
+    ],
+    "hdmap:levelOfDetail": "crosswalk",
+    "hdmap:trafficDirection": "right-hand"
+  },
+  "hdmap:quantity": {
+    "@type": "hdmap:Quantity",
+    "hdmap:length": {
+      "@value": "1,46",
```
Comment (lenasauermann) [inline hdmap/hdmap-instance.json:96]: decide on one: . or ,

**Review by lenasauermann — CHANGES_REQUESTED** [review]

File: hdmap/hdmap-instance.json (hunk 17 above)
Comment (3Dbastian) [reply hdmap/hdmap-instance.json]: . 
changed it

**Review by MircoNierenz — COMMENTED** [review]
ok for me

**Review by MircoNierenz — APPROVED** [review]
ok for me

File: HD-Map/hdmap_ontology.ttl (hunk 4 above)
[the remark this answers is not in the document]
Comment (robertschubert) [reply HD-Map/hdmap_ontology.ttl]: reopended because HdMap is hdmap again.

**Review by robertschubert — CHANGES_REQUESTED** [review]
reopened one review comment since HdMap is hdmap again.
After this we are ready to merge.

**Comment by 3Dbastian** [comment]
> reopened one review comment since HdMap is hdmap again. After this we are ready to merge.

fixed

File: HD-Map/hdmap_ontology.ttl (hunk 4 above)
[the remark this answers is not in the document]
Comment (robertschubert) [reply HD-Map/hdmap_ontology.ttl]: resolved

**Review by robertschubert — APPROVED** [review]
HdMap fixed, approved

**Comment by robertschubert** [comment]
all comments are resolved, we have 3 approvals, merge.
Merge to main.

**Pull request merged**